pkg/grpc/grpc.go

```diff
@@ -108,31 +109,21 @@ func (g *Manager) GetAppChannel() (channel.AppChannel, error) {
 // GetAppClient returns the gRPC connection to the local app.
 // If there's no active connection to the app, it creates one.
 func (g *Manager) GetAppClient() (grpc.ClientConnInterface, error) {
-	if g.localConnCreateFn != nil {
-		return g.localConn.Get(g.localConnCreateFn)
-	}
-	return g.localConn.Get(g.defaultLocalConnCreateFn)
-}
+	if g.appClientConn == nil {
```

Comment:
This section of the code cannot have concurrent access, thus no usage of locks.

Reply:
I think the reason for the lock was because otherwise `go test -race` would fail otherwise, either here or in the runtime. Can you please confirm that both this package (pkg/grpc) and pkg/runtime pass with `go test -race` without the lock?

Reply:
I do see this method invoked in multiple places, so a lock may be needed to avoid race conditions

Reply:
See comment below